src/Jupyter/Extensions.cs

```diff
@@ -75,71 +76,85 @@ public static string ToHumanReadableBytes(this long nBytes)
         }
 
         /// <summary>
-        ///     Adds functionality to a given quantum simulator to display
-        ///     diagnostic output with rich Jupyter formatting.
+        ///     An enumerable source of the significant amplitudes of this state
+        ///     vector and their labels, where significance and labels are
+        ///     defined by the values loaded from <paramref name="configurationSource" />.
         /// </summary>
-        /// <param name="simulator">
-        ///     The simulator to be augmented with Jupyter display
-        ///     functionality.
-        /// </param>
-        /// <param name="channel">
-        ///     The Jupyter display channel to be used to display diagnostic
-        ///     output.
-        /// </param>
-        /// <param name="configurationSource">
-        ///      A source of configuration options to be used to set display
-        ///      preferences. Typically, this will be provided by the service
-        ///      provider configured when an execution engine is constructed.
-        /// </param>
-        /// <returns>
-        ///     The value of <paramref name="simulator" />.
-        /// </returns>
-        public static QuantumSimulator WithJupyterDisplay(this QuantumSimulator simulator, IChannel channel, IConfigurationSource configurationSource)
-        {
-            // First, we disable console-based logging so as to not
-            // duplicate messages.
-            simulator.DisableLogToConsole();
-            // Next, we attach the display channel's standard output handling
-            // to the log event.
-            simulator.OnLog += channel.Stdout;
+        public static IEnumerable<(Complex, string)> SignificantAmplitudes(
+            this CommonNativeSimulator.DisplayableState state, 
+            IConfigurationSource configurationSource
+        ) => state.SignificantAmplitudes(
+            configurationSource.BasisStateLabelingConvention,
+            configurationSource.TruncateSmallAmplitudes,
+            configurationSource.TruncationThreshold
+        );
 
-            // Next, we register the generic version of the DumpMachine callable
-            // as an ICallable with the simulator. Below, we'll provide our
-            // implementation of DumpMachine with the channel and configuration
-            // source we got as arguments. At the moment, there's no direct
-            // way to do this when registering an implementation, so we instead
-            // get an instance of the newly registered callable and set its
-            // properties accordingly.
-            simulator.Register(
-                typeof(Diagnostics.DumpMachine<>), typeof(JupyterDumpMachine<>),
-                signature: typeof(ICallable)
-            );
+        // /// <summary>
```

Comment:
Would be good to remove this commented code now that it's no longer needed to workaround the old-style dumps in qsharp-runtime.